crypto/evp/p_rsa.c

```diff
@@ -405,6 +405,10 @@ static int pkey_rsa_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2) {
         OPENSSL_PUT_ERROR(EVP, EVP_R_ILLEGAL_OR_UNSUPPORTED_PADDING_MODE);
         return 0;
       }
+      if (p1 != RSA_PKCS1_PSS_PADDING && ctx->pmeth->pkey_id == EVP_PKEY_RSA_PSS) {
```

Comment:
Should we do something about that logic above your if statement?

Reply:
Maybe no.
The logic above is to check if the padding mode and the corresponding operation are compatible.
This line is to make sure the padding should be pss only when pmeth is rsassaPss (based on RFC). Kind of checking if padding and key type are compatible.

Reply:
Sure, I just found the if statement confusing.